WeScan/Scan/ScannerViewController.swift

```diff
@@ -177,19 +180,98 @@ final class ScannerViewController: UIViewController {
+                device.exposureMode = .continuousAutoExposure
+            }
+            
+            device.unlockForConfiguration()
+        } catch {
+            let error = ImageScannerControllerError.inputDevice
+            guard let captureSessionManager = captureSessionManager else { return }
+            captureSessionManager.delegate?.captureSessionManager(captureSessionManager, didFailWithError: error)
+            return
+        }
+        
+    }
+    
     @objc private func captureImage(_ sender: UIButton) {
         (navigationController as? ImageScannerController)?.flashToBlack()
         shutterButton.isUserInteractionEnabled = false
         captureSessionManager?.capturePhoto()
     }
     
     @objc private func toggleAutoScan() {
-        if CaptureSession.current.autoScanEnabled {
-            CaptureSession.current.autoScanEnabled = false
+        if CaptureSession.current.isAutoModeEnabled {
+            CaptureSession.current.isAutoModeEnabled = false
             autoScanButton.title = NSLocalizedString("wescan.scanning.manual", tableName: nil, bundle: Bundle(for: ScannerViewController.self), value: "Manual", comment: "The manual button state")
         } else {
-            CaptureSession.current.autoScanEnabled = true
+            CaptureSession.current.isAutoModeEnabled = true
             autoScanButton.title = NSLocalizedString("wescan.scanning.auto", tableName: nil, bundle: Bundle(for: ScannerViewController.self), value: "Auto", comment: "The auto button state")
+            
+            /// Reset the focus to continousAutoFocus
+            do {
+                guard let device = AVCaptureDevice.default(for: AVMediaType.video) else { return }
+                try device.lockForConfiguration()
+            
+                if device.isFocusPointOfInterestSupported && device.isFocusModeSupported(.continuousAutoFocus) {
+                    device.focusMode = .continuousAutoFocus
+                }
+                
+                if device.isExposurePointOfInterestSupported, device.isExposureModeSupported(.continuousAutoExposure) {
+                    device.exposureMode = .continuousAutoExposure
+                }
+                
+                device.unlockForConfiguration()
+            } catch {
+                let error = ImageScannerControllerError.inputDevice
+                guard let captureSessionManager = captureSessionManager else { return }
+                captureSessionManager.delegate?.captureSessionManager(captureSessionManager, didFailWithError: error)
+                return
+            }
+            
+            /// Remove the focus rectangle if one exists
+            if let focusRectangle = focusRectangle {
+                UIView.animate(withDuration: 0.3, delay: 1.0, animations: {
+                    focusRectangle.alpha = 0.0
+                }, completion: { (_) in
+                    focusRectangle.removeFromSuperview()
+                })
```

Comment:
Again I think we can make this one shorter.

Reply:
Resolved: 9d1e1e4